src/main/scala/io/circe/bson/BsonCodecInstances.scala

```diff
@@ -55,7 +55,11 @@ trait BsonCodecInstances {
     case BSONJavaScriptWS(value) => Right(Json.fromString(value))
     case BSONMaxKey              => Left(readerFailure(bson))
     case BSONMinKey              => Left(readerFailure(bson))
-    case BSONObjectID(_)         => Left(readerFailure(bson))
+    case BSONObjectID(id)        =>
+      BSONObjectID.parse(BSONObjectID(id).stringify) match {
```

Comment:
Can you explain why it's necessary to extract the ID, parse it, and extract it again?

Reply:
changed it!